g4f/gui/client/index.html

```diff
@@ -66,6 +66,121 @@
         window.share_id = "{{share_id}}";
         window.share_url = "{{share_url}}";
         window.start_id = "{{conversation_id}}";
+        
```

Comment:
Why you adding only lines and don't remove the old code?
And why so many lines? Did you wrote it and why?
Provider selection does work perfect for me?
You could rewrite the select boxes and add maybe a cache there?